SU-HSDO/suhumsci
# STN-661: Carousel Height Refinement

# [STN-661](https://sparkbox.atlassian.net/browse/STN-661) READY FOR REVIEW

## Summary
Part One:

The work in this card refines the height of text boxes in the Carousel. JS is used to find the height of each text area and identify the tallest. The tallest text area is used to update all containers to match that height.

Part Two:

For site owners that would prefer the existing behavior a utility class called `hb-carousel-reset-height` has been created to prevent the JS from running for that carousel (or any carousels in the same container).

Notes: 
* It was determined not to include this work within the experimental toggle. Reference [slack thread](https://sparkbox.slack.com/archives/CM9J5THPV/p1613675494110900)

## Need Review By (Date)
2/26/2021

## Urgency
medium

## Steps to Test
1. In your CLI, run `npm run test` and confirm all tests pass. Then run `npm run build` to build the CSS and JS.
2. To confirm the height of the text boxes in carousels are all set to the same height as the carousel with the tallest text box height, go to http://sparkbox-sandbox.suhumsci.loc/carousel-hero-image. You if do not have this page locally, you can sync with staging or create a new flexible page and add a carousel with multiple slides in the hero area and again within the main content section. If you sync with staging, note that you may need to upload new images for each carousel.
3. - [x] Confirm the text boxes on multiple slides within a single carousel are the same height on both Colorful and Traditional themes.
4. - [x] Resize the width of the browser window. Confirm the text boxes resize according the tallest box with the most content. Note the slick dots and arrows have always had some alignment funkiness when the page is resized. This is pre-existing.
5. - [x] Test new utility class of `hb-carousel-reset-height`. Add `hb-carousel-reset-height` to a carousel component via Layout Builder. Confirm that the height of the text boxes for each slide within a carousel defaults to fit the content rather than the tallest text box.
6. - [x] Confirm that the JavaScript only runs on pages with carousels.
7.  - [x] Confirm there are not JavaScript errors from the work done in this PR. (There are some errors re: fontawesome and a pre-existing  Uncaught TypeError.)
8. - [x] Confirm the work in this PR does not impact other hero elements.
9. - [ ] Confirm the new utility class of `hb-carousel-reset-height` has been added to the docs.
